massa-execution-worker/src/tests/scenarios_mandatories.rs

```diff
@@ -321,6 +321,164 @@ fn send_and_receive_async_message() {
+    let operation = create_execute_sc_operation(&keypair, bytecode, datastore).unwrap();
+    storage.store_operations(vec![operation.clone()]);
+    let block = create_block(keypair, vec![operation], Slot::new(1, 0)).unwrap();
+    // store the block in storage
+    storage.store_block(block.clone());
+
+    // set our block as a final block so the message is sent
+    let mut finalized_blocks: HashMap<Slot, BlockId> = Default::default();
+    finalized_blocks.insert(block.content.header.content.slot, block.id);
+    let mut block_storage: PreHashMap<BlockId, Storage> = Default::default();
+    block_storage.insert(block.id, storage.clone());
+    blockclique_blocks.insert(block.content.header.content.slot, block.id);
+    controller.update_blockclique_status(
+        finalized_blocks.clone(),
+        Some(blockclique_blocks.clone()),
+        block_storage.clone(),
+    );
+    // sleep for 150ms to reach the message execution period
+    std::thread::sleep(Duration::from_millis(10));
+
+    // retrieve events emitted by smart contracts
+    let events = controller.get_filtered_sc_output_event(EventFilter {
+        ..Default::default()
+    });
+
+    println!("events: {:?}", events);
+
+    // match the events
+    assert!(events.len() == 2, "Two event was expected");
+    assert_eq!(events[0].data, "Triggered");
+
+    // keypair associated to thread 1
+    let keypair = KeyPair::from_str("S1kEBGgxHFBdsNC4HtRHhsZsB5irAtYHEmuAKATkfiomYmj58tm").unwrap();
+    // load bytecode
+    // you can check the source code of the following wasm file in massa-unit-tests-src
+    let bytecode = include_bytes!("./wasm/send_message_wrong_trigger.wasm");
+    let datastore = BTreeMap::new();
+
+    // create the block containing the smart contract execution operation
+    let operation = create_execute_sc_operation(&keypair, bytecode, datastore).unwrap();
+    storage.store_operations(vec![operation.clone()]);
+    let block = create_block(keypair, vec![operation], Slot::new(1, 1)).unwrap();
+    // store the block in storage
+    storage.store_block(block.clone());
+
+    // set our block as a final block so the message is sent
+    finalized_blocks.insert(block.content.header.content.slot, block.id);
+    let mut block_storage: PreHashMap<BlockId, Storage> = Default::default();
+    block_storage.insert(block.id, storage.clone());
+    blockclique_blocks.insert(block.content.header.content.slot, block.id);
+    controller.update_blockclique_status(finalized_blocks.clone(), None, block_storage.clone());
+    // sleep for 150ms to reach the message execution period
+    std::thread::sleep(Duration::from_millis(10));
+
+    // retrieve events emitted by smart contracts
+    let events = controller.get_filtered_sc_output_event(EventFilter {
+        ..Default::default()
+    });
+
+    // match the events
+    assert!(events.len() == 2, "Two event was expected");
+    assert_eq!(events[0].data, "Triggered");
```

Comment:
Should we check the other events as well?

Reply:
Fixed in : https://github.com/massalabs/massa/pull/3228